### PR TITLE
fix kubernetes with traefik

### PR DESCRIPTION
**Pull request by tomaszduda23**
Address https://github.com/acouvreur/sablier/issues/142, https://github.com/acouvreur/sablier/issues/143, https://github.com/acouvreur/sablier/issues/131 and https://github.com/acouvreur/sablier/issues/62

### CODE REVIEW

**Comment by acouvreur** [comment]
Hi @tomaszduda23 thank you for your contribution, I'll look into the details soon!

**Comment by acouvreur** [comment]
After a quick reading, here's what I understand of your changes:

1. Sablier states that the backend is ready
2. You issue a request yourself to see if the underlying backend is in fact ready (sync between observed state and actual state)
3. If it's not ready
    1. If it's in the case of a blocking request you issue a redirect
    2. If it's in the case of a dynamic request you serve the response from sablie

Makes sense to me!
However I'll need your input and description on what you've done so we're sure everything is correct.

Also, check the builds failure, some tests appears to be broken. You might want to add test to cover all use cases you added.

Thanks for your contribution!

**Comment by acouvreur** [comment]
And it seems like the integration test for e2e worked!
image

Good job!

**Comment by tomaszduda23** [comment]
> * Sablier states that the backend is ready
> 
> * You issue a request yourself to see if the underlying backend is in fact ready (sync between observed state and actual state)
> 
> * If it's not ready
>   
>   1. If it's in the case of a blocking request you issue a redirect
>   2. If it's in the case of a dynamic request you serve the response from sablie

The request is issued to check if 503 was generated inside traefik due to missing server or by server itself. Traefik is not able to attach server to existing router. When scaling from 0 to 1 client has to make one extra request. Otherwise there will be 503.

**Comment by tomaszduda23** [comment]
Tests are fixed. The rest of fails in pipeline seems to be related to permissions.

File: plugins/traefik/main_test.go [hunk 1]
```diff
@@ -61,6 +67,90 @@ func TestSablierMiddleware_ServeHTTP(t *testing.T) {
...
+					httptrace.ContextClientTrace(r.Context()).WroteHeaders()
+					fmt.Fprint(w, "response from service")
+				}),
+				Config: &Config{
+					SessionDuration: "1m",
+					Blocking:        &BlockingConfiguration{},
+				},
+			},
+			expected: "response from sablier",
+			// is this correct for blocking? I would expect to get error
```
Comment (tomaszduda23) [inline plugins/traefik/main_test.go:132]: This seems to be wired to return 200 and json but this is how it was working before as well.

**Comment by acouvreur** [comment]
I would like to merge this into beta, can you try and rebase your branch into beta first ?

**Comment by tomaszduda23** [comment]
done

**Pull request merged**